avalon/tools/workfiles/app.py

```diff
@@ -748,7 +749,7 @@ def __init__(self, parent=None):
         widgets = {
             "pages": QtWidgets.QStackedWidget(),
             "body": QtWidgets.QWidget(),
-            "assets": AssetWidget(silo_creatable=False),
+            "assets": AssetWidget(parent=self),
```

Comment:
As asked before - what's changed that now we need to pass the parent?

Reply:
Same answer as before, but not sure about effect here, because I prepared this branch just in time workfiles + context manager PR was merged, so I didn't have time to verify if this solved stylesheet issue.
Now I can't verify, because home-office means no applications for me.

As I look into code now, I'm 90% sure it is not right... But it seems impossible to solve parenting here due to new widgets storing way.
```
pages = {
    "home": QtWidgets.QWidget()
}

widgets = {
    "pages": QtWidgets.QStackedWidget(), # parent=pages ["home"]
    "body": QtWidgets.QWidget(),         # parent=widgets["pages"]
    "assets": AssetWidget(parent=self),  # parent=widgets["body"]
    "tasks": TasksWidget(),              # parent=widgets["body"]
    "files": FilesWidget()               # parent=widgets["body"]
}
```

Reply:
Are these widgets not part of a layout?

The only time you need to explicitly provide a parent is when it's a standalone widget. Whenever you call `layout.addWidget(widget)` then `widget` is assigned the layout as a parent, which itself is child to another widget, most likely the window.

If not, you can also call [`widget.setParent`](https://doc.qt.io/qt-5/qwidget.html#setParent) after instantiation.

Reply:
Not really, mainly PySide have issues with parenting and propagation of styles when parent is not explicitly set and should be propagated internally. And other (I think it was PyQt4, not sure) had issues when stylesheet is set after `show` without understandable reason, stylesheets for specific `objectName` worked, but not for it's children.
e.g. `#MainWidget QLabel {backgroud: red;}` had no effect for Qlabels in `MainWidget`.

You got true, `setParent` should help here.